### PR TITLE
Provide means to track API limit request info

### PR DESCRIPTION
**Pull request by kendagriff**
So, the library – as is – is not particularly amenable to collecting additional data from Salesforce's REST responses – `request`, for example, returns the `:body` parameter to all the public-facing functions (e.g. `so->get`, `so->create`, etc.).

But, the response still contains other important information, such as API limit information. This may not be the best implementation, because it's stateful, but to the client user, it's dead simple:

```
(so->get ...)
```

Calling the function now returns the body of the response, as usual, but in calling it, it's also recorded the limit information from the response header to an atom: `limit-info`. The client user can retrieve the used and available API counts by calling:

```
(read-limit-info)
=> {:used 100 :available 15000}
```

Again, a little different, but simple to implement, and easy to use, given the current design of the library.

### CODE REVIEW

**Comment by owainlewis** [comment]
Hi. Thanks for the pull request. I don't have time to properly review but will try and get round to it sometime next week. Thanks :+1:

**Comment by sventech** [comment]
Bump. This looks like a very good idea and is fairly well implemented.

File: project.clj [hunk 1]
```diff
@@ -1,8 +1,8 @@
-(defproject salesforce "1.0.1"
+(defproject com.banzai/salesforce "1.0.6"
```
Comment (owainlewis) [inline project.clj]: Need to update this back to salesforce

**Comment by kendagriff** [comment]
Ah, shoot. Must have leaked in from my private branch. Lemme see if I can't fix that...

**Comment by kendagriff** [comment]
@owainlewis Do you mind if I simply push a commit fixing the problem, rather than trying to undo the public history?

**Comment by owainlewis** [comment]
@kendagriff no that sounds fine to me :+1:  Thanks

**Comment by kendagriff** [comment]
Fixed and pushed.

**Comment by owainlewis** [comment]
Thanks @kendagriff

**Pull request merged**